src/rust/src/test_support.rs

```diff
@@ -103,6 +103,57 @@ fn pkcs7_verify(
     Ok(())
 }
 
+#[cfg(not(CRYPTOGRAPHY_IS_BORINGSSL))]
```

Comment:
What is this needed for? We have public APIs for PKCS#7 encryption

Reply:
I coded this for testing purposes: to encode using another algorithm, so that we can check the algorithm is unsupported (for coverage, mostly).

For most of the tests, I can replace the `test_support.pkcs7_encrypt` with `PKCS7EnvelopeBuilder`, but not for `test_smime_decrypt_algorithm_not_supported`.

Another solution can be to store an AES-256-CBC encoded PKCS7, and read it during the tests, if you prefer 😄

Reply:
For test cases, is there a reason not to pre-generate them and then just check them into the vectors folder?

Reply:
No, there's none, i can definitely do it! 

Also, does that mean I should get rid of the `test_support.pkcs7_decrypt` function and use the newly developed one?

Reply:
If we pre-generate the test cases, we actually don't need the generator code checked in at all.

Reply:
The new commits should remove the `test_support.pkcs_encyrpt` function and replace its usage with new vectors (for which I'll do a separate PR today).

I'm talking about the previous tests for the `PKCS7EnvelopeBuilder`, that are using `test_support.pkcs7_decrypt`. I assume I should replace this function with the new public API we're developing, `pkcs7_decrypt_{encoding}`?

Reply:
Ahhhh, sorry I misunderstood. Yes, I think replacing those with the new public API would be good.

Reply:
Will do today too :)

Reply:
Done!